lib/magma/model.rb

```diff
@@ -238,8 +249,16 @@ def inherited(magma_model)
         )
 
         super
-        magma_model.attribute(:created_at, type: DateTime, hide: true)
-        magma_model.attribute(:updated_at, type: DateTime, hide: true)
+        magma_model.send(:attribute, :created_at, type: DateTime, hide: true)
+        magma_model.send(:attribute, :updated_at, type: DateTime, hide: true)
+      end
+
+      private
+
+      # basic attribute, holds any pg data type
+      def attribute(attr_name, opts = {})
+        klass = opts.delete(:attribute_class) || Magma::Attribute
```

Comment:
might be better to start requiring attribute_class as a method argument rather than passing it in opts since we want to deprecate using Magma::Attribute as a base class (i.e., attribute_class will be required).